sklearn/_build_utils/openmp_helpers.py

```diff
@@ -0,0 +1,146 @@
+"""Helpers for OpenMP support during the build."""
+
+# This code is adapted for a large part from the astropy openmp helpers, which
+# can be found at: https://github.com/astropy/astropy-helpers/blob/master/astropy_helpers/openmp_helpers.py  # noqa
+
+
+import os
+import sys
+import glob
+import tempfile
+import textwrap
+import subprocess
+
+from numpy.distutils.ccompiler import new_compiler
+from distutils.sysconfig import customize_compiler
+from distutils.errors import CompileError, LinkError
+
+
+CCODE = textwrap.dedent(
+    """\
+    #include <omp.h>
+    #include <stdio.h>
+    int main(void) {
+    #pragma omp parallel
+    printf("nthreads=%d\\n", omp_get_num_threads());
+    return 0;
+    }
+    """)
+
+
+def get_openmp_flag(compiler):
+    if os.getenv('SKLEARN_NO_OPENMP'):
```

Comment:
I think it would be clearer to the reader if this check was just done in `setup.py`. Right now it's a bit surprising that this is something performed by `get_openmp_flag`